Mods/SML/Source/SML/Public/Toolkit/KismetBytecodeDisassembler.h

```diff
@@ -5,21 +5,22 @@
 
 class SML_API FSMLKismetBytecodeDisassembler {
 public:
-	/** Converts a single expression into json object */
-	TSharedPtr<FJsonObject> SerializeExpression(int32& ScriptIndex);
-
 	/** Parses a block of statements until it hits return */
 	TArray<TSharedPtr<FJsonValue>> SerializeFunction(UStruct* Function);
 
-	/** Computes length of the statement in bytes and returns it. Returns false if given index does not correspond to any statement (e.g if it is inside of some statement) */
-	bool GetStatementLength(UStruct* Function, int32 StatementIndex, int32& OutStatementLength);
+	/** Converts a single statement at the given index into json object. Will throw an error if the statement index does not correspond to the start of a statement.*/
+	TSharedPtr<FJsonObject> SerializeStatement(UStruct* Function, int32 StatementIndex);
 
 	/** Returns index of the first statement using given opcode */
 	bool FindFirstStatementOfType(UStruct* Function, int32 StartIndex, uint8 StatementOpcode, int32& OutStatementIndex);
+
 private:
 	TWeakObjectPtr<UClass> SelfScope;
 	TArray<uint8> Script;
 
+	/** Converts a single expression into json object, provided Script and SelfScope have been set */
+	TSharedPtr<FJsonObject> SerializeExpression(int32& ScriptIndex);
```

Comment:
This was part of the API, why make it private?

Reply:
As I mentioned in Discord, this will crash if called before one of the other public functions is called because the Script and SelfScope member variables aren't set.  It's really an internal utility function that shouldn't have been made public, IMO.  That said, after we discussed, I agreed to move it back for now.  If I clean up the API of this class in a future PR, I can address it then.

Reply:
Ack